benches/hierarchy/engine/mod.rs

```diff
@@ -14,7 +14,7 @@ fn bench_engine_put<E: Engine, F: EngineFactory<E>>(
 ) {
     let engine = config.engine_factory.build();
     let ctx = Context::new();
-    bencher.iter_with_setup(
+    bencher.iter_batched(
```

Comment:
Prefer the `_ref` version because we simply want to test how long it takes for `put` and we don't want to include the time of `Drop`.

Reply:
Every setup function has some kind of object creation. I replace all `iter_batched` with `iter_batched_ref.`